app/javascript/reducers/detailsReducers.js

```diff
@@ -104,6 +107,30 @@ function fetchDetails(
       }
     }
 
+    case actionTypes.HANDLE_EMAIL_INPUT_CHANGE: {
+      const email = action.payload.value
+      const validateEmail = /^[a-zA-Z0-9_!#$%&’*+/=?`'{^.-]*@(?:[A-Z0-9-]+\.)+[A-Z]{2,6}$/i.test(email)
+      validateEmail ? (state.emailErrorMessage = '') : (state.emailErrorMessage = 'Please enter a valid email.')
+      state.emailErrorMessage === '' ? (state.disableActionBtn = false) : (state.disableActionBtn = true)
```

Comment:
for 113 and 114, it seems tangled, as yore using 113's result to perform the logic on 114, but they really are the same case.
 I suggest direct logic to make it more clear...
```suggestion
      if (validateEmail(email) {
        state.emailErrorMessage = ''
        state.disableActionBtn = false
      } else {
         state.emailErrorMessage = 'Please enter a valid email.'
         state.disableActionBtn = true
       }
```
Or, consider putting the emailErrorMessage directly on the component render, but hiding that when disableActionBtn state is true.  Might eliminate this code altogether.